File: app/src/androidTest/java/com/leaguechampions/robots/SettingsRobot.java
package com.leaguechampions.robots;

import android.support.test.InstrumentationRegistry;

import com.leaguechampions.BuildConfig;
import com.leaguechampions.R;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
import static android.support.test.espresso.matcher.ViewMatchers.withText;

public class SettingsRobot {

    public SettingsRobot checkVersionIsDisplayed() {
        onView(withText(String.format(InstrumentationRegistry.getTargetContext().getString(R.string.version), BuildConfig.VERSION_NAME)))
                .check(matches(isDisplayed()));
        return this;
    }
}

File: app/src/androidTest/java/com/leaguechampions/ui/settings/SettingsRobot.kt
package com.leaguechampions.ui.settings

import androidx.test.InstrumentationRegistry
import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
import androidx.test.espresso.matcher.ViewMatchers.withText
import com.leaguechampions.BuildConfig
import com.leaguechampions.R

class SettingsRobot {

    fun checkVersionIsDisplayed(): SettingsRobot {
        onView(withText(String.format(InstrumentationRegistry.getTargetContext().getString(R.string.version), BuildConfig.VERSION_NAME)))
                .check(matches(isDisplayed()))
        return this
    }
}
